prov/rxm/src/rxm.h

```diff
@@ -474,6 +473,16 @@ struct rxm_tx_rndv_buf {
 	struct rxm_pkt pkt;
 };
 
+union rxm_tx_buf {
+	struct {
+		struct rxm_buf hdr;
+		enum rxm_buf_pool_type type;
+	} top;
+	struct rxm_tx_base_buf base;
+	struct rxm_tx_eager_buf eager;
+	struct rxm_tx_sar_buf sar;
+	struct rxm_tx_rndv_buf rndv;
+};
```

Comment:
Is the union needed?

Reply:
This is needed only in `rxm_tx_buf_release` and `rxm_tx_buf_get`. I'm looking how to avoid this